fix: Do not try to build sse3 code if the target feature is missing

lu-zero opened the pull request:
It should tackle the main annoyance of #4 and also prevent sigills if for whichever reason sse3 disabled on x86_64.

It will require to build passing an appropriate `-Ctarget-cpu`.

lu-zero committed:
fix: Do not try to build sse3 code if the target feature is missing

FlorentinDUBOIS commented on src/protocol/utils.rs:
```diff
@@ -119,7 +119,7 @@ macro_rules! compile_lookup {
 
             #[inline]
-            #[cfg(feature="simd")]
+            #[cfg(all(feature="simd", target_feature="sse3"))]
```
**High — Gate on the architecture and strongest ISA requirement**

This predicate is too weak for the block it enables. The implementation imports `std::arch::x86_64`, but `target_feature="sse3"` can also be true on 32-bit x86: `RUSTFLAGS='-C target-feature=+sse3' cargo +stable check --offline --target i686-unknown-linux-musl --all-features` fails here with `E0432` because `std::arch::x86_64` is unavailable. More importantly, the enabled path calls `_mm_cmpestri`, which Rust marks `#[target_feature(enable = "sse4.2")]`; an `sse3,-sse4.2` probe still emits `pcmpestri`, so an SSE3-only CPU can retain the `SIGILL` this PR is meant to prevent. Please gate all four SIMD sites on `target_arch = "x86_64"` and `target_feature = "sse4.2"` (or use runtime dispatch), and make the scalar branch the exact negation of that predicate.

FlorentinDUBOIS reviewed (changes requested):
Automated cross-review (correctness / Rust cfg / guidelines / simplification / security + adversarial checker). One verified finding is inline; comments batched into one review.

FlorentinDUBOIS committed:
fix(parser): require SSE4.2 for SIMD lookup

Guard SIMD parser selection by both x86_64 and SSE4.2 so unsupported architectures and CPUs use the scalar fallback instead of failing to compile or executing unsupported instructions.

Document the fallback behavior for consumers.

Signed-off-by: Florentin Dubois <florentin.dubois@clever.cloud>

FlorentinDUBOIS committed:
test: satisfy strict Clippy validation

Update parser tests to use complete writes, supported borrowed iteration, and lint-clean benchmark barriers so the rebased PR passes the strict all-target pre-push gate without suppressing diagnostics.

Signed-off-by: Florentin Dubois <florentin.dubois@clever.cloud>
